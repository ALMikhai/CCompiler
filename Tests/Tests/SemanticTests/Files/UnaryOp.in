int f() {
    int a = -1;
    int b = +a;
    int c = !a;
    int d = ~2;
    float a1 = 0.0;
    float *b1 = &a1;
    float c1 = *b1;
    float **b2 = &b1;
}
