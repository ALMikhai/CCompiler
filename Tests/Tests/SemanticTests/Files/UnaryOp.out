\-TRANSLATIONUNIT
  \-FUNC
    |-DECLSPEC
    | |-INT
    |-INITDECLARATOR
    | \-DECLARATOR
    |   \-FUNC
    |     |-f
    |     \-EMPTY
    \-COMPOUNDSTAT
      |-DECLLIST
      | |-DECL
      | | |-DECLSPEC
      | | | |-INT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-a
      | |     \-SUB
      | |       \-1
      | |-DECL
      | | |-DECLSPEC
      | | | |-INT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-b
      | |     \-ADD
      | |       \-a
      | |-DECL
      | | |-DECLSPEC
      | | | |-INT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-c
      | |     \-NOT
      | |       \-a
      | |-DECL
      | | |-DECLSPEC
      | | | |-INT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-d
      | |     \-TILDE
      | |       \-2
      | |-DECL
      | | |-DECLSPEC
      | | | |-FLOAT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-a1
      | |     \-0
      | |-DECL
      | | |-DECLSPEC
      | | | |-FLOAT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | |-POINTER
      | |     | \-b1
      | |     \-BITAND
      | |       \-a1
      | |-DECL
      | | |-DECLSPEC
      | | | |-FLOAT
      | | \-INITDECLARATORLIST
      | |   \-INITDECLARATOR
      | |     |-DECLARATOR
      | |     | \-c1
      | |     \-MULT
      | |       \-b1
      | \-DECL
      |   |-DECLSPEC
      |   | |-FLOAT
      |   \-INITDECLARATORLIST
      |     \-INITDECLARATOR
      |       |-DECLARATOR
      |       | |-POINTER
      |       | | \-POINTER
      |       | \-b2
      |       \-BITAND
      |         \-b1

Structs { }
Symbols {
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: printf
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: scanf
	FUNC returning INT
	Arguments{ }
	Structs { }
	Symbols {
		INT :: a
		INT :: b
		INT :: c
		INT :: d
		FLOAT :: a1
		POINTER to FLOAT :: b1
		FLOAT :: c1
		POINTER to POINTER to FLOAT :: b2
	} :: f
}
